tiledb/sm/query/reader.cc

```diff
@@ -877,6 +880,43 @@ Status Reader::check_subarray() const {
   return Status::Ok();
 }
 
+Status Reader::check_buffer_sizes() const {
```

Comment:
This is only checking the validity buffer? Should we rename it `check_validity_buffer_sizes`?

Reply:
Done. There's a similarly named `check_buffer_sizes` in the `Writer` that I was mirroring here, but we can change that later if we extend the functionality to check something other than validity buffers.